fem/libceed/diffusion.h

```diff
@@ -0,0 +1,199 @@
+// Copyright (c) 2010, Lawrence Livermore National Security, LLC. Produced at
+// the Lawrence Livermore National Laboratory. LLNL-CODE-443211. All Rights
+// reserved. See file COPYRIGHT for details.
+//
+// This file is part of the MFEM library. For more information and source code
+// availability see http://mfem.org.
+//
+// MFEM is free software; you can redistribute it and/or modify it under the
+// terms of the GNU Lesser General Public License (as published by the Free
+// Software Foundation) version 2.1 dated February 1999.
+
+/// A structure used to pass additional data to f_build_diff and f_apply_diff
+struct BuildContext { CeedInt dim, space_dim; CeedScalar coeff; };
+
+/// libCEED Q-function for building quadrature data for a diffusion operator
```

Comment:
Can't these Q-functions reuse code somehow? The two functions in this file are nearly identical.

Reply:
I don't see how, but if you have a proposition I would be interested to know.

Reply:
I think using the `CEED_QFUNCTION` macro would work to create functions, however I don't really see how to factorize code.